roxctl/scanner/generate/generate.go

```diff
@@ -5,57 +5,103 @@ import (
 	"fmt"
 	"net/http"
 	"strings"
+	"time"
 
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/apiparams"
+	"github.com/stackrox/rox/pkg/errox"
 	"github.com/stackrox/rox/pkg/istioutils"
+	"github.com/stackrox/rox/roxctl/common/environment"
 	"github.com/stackrox/rox/roxctl/common/flags"
 	"github.com/stackrox/rox/roxctl/common/zipdownload"
 	"github.com/stackrox/rox/roxctl/scanner/clustertype"
 )
 
-// Options stores options related to scanner generate commands.
-type Options struct {
-	OutputDir string
+const (
+	scannerGenerateAPIPath = "/api/extensions/scanner/zip"
+
+	istioSupportArg = "istio-support"
+)
+
+type scannerGenerateCommand struct {
+	// Properties that are bound to cobra flags.
+	outputDir string
+	apiParams apiparams.Scanner
+	timeout   time.Duration
+
+	// Properties that are injected or constructed.
+	env environment.Environment
+}
+
+func (cmd *scannerGenerateCommand) construct(c *cobra.Command) {
+	cmd.timeout = flags.Timeout(c)
+}
+
+func (cmd *scannerGenerateCommand) validate() error {
+	// Validate supported Istio versions.
+	if cmd.apiParams.IstioVersion != "" {
+		for _, istioVersion := range istioutils.ListKnownIstioVersions() {
+			if cmd.apiParams.IstioVersion == istioVersion {
+				return nil
+			}
+		}
+
+		return errox.NewErrInvalidArgs(fmt.Sprintf(
```

Comment:
Prefer `errox.InvalidArgs.Newf()` instead.

Reply:
@dhaus67 I have tried that initially, but the problem was that the output message wasn't prefixed with `invalid arguments`. And with `errox.NewErrInvalidArgs` it was.

Reply:
If you want to keep the `invalid arguments` part, you shall use `.CausedBy()`. However, I'm not sure `invalid arguments` part adds any value.